lib/ecto/adapters/postgres/connection.ex

```diff
@@ -702,6 +719,19 @@ if Code.ensure_loaded?(Postgrex.Connection) do
     defp null_expr(true), do: "NULL"
     defp null_expr(_), do: []
 
+    defp new_constraint_expr(%Ecto.Migration.Constraint{check: nil, exclude: nil})  do
```

Comment:
Oops, I didn't see your comment when I added mine. Just to clarify: I think those two clauses should go because you already check for them in the runner.